diff --git a/node_modules/vue-loader/lib/loaders/pitcher.js b/node_modules/vue-loader/lib/loaders/pitcher.js
index 222b913..f6834bf 100644
--- a/node_modules/vue-loader/lib/loaders/pitcher.js
+++ b/node_modules/vue-loader/lib/loaders/pitcher.js
@@ -163,5 +163,17 @@ module.exports.pitch = function (remainingRequest) {
   // both that rule and the cloned rule will match, resulting in duplicated
   // loaders. Therefore it is necessary to perform a dedupe here.
   const request = genRequest(loaders)
-  return `import mod from ${request}; export default mod; export * from ${request}`
+  const allPages = process.env.allPages;
+  let relativePath = require('path').relative(process.env.context, this.resourcePath);
+  relativePath = relativePath.replace(/\.\w+$/g, '');
+  let injectCode = '';
+  if (relativePath === 'App') {
+    // 目前是手动 createApp
+  } else if (allPages.includes(relativePath)) {
+    injectCode = `mod.render = function(){}; wx.createPage(mod)`
+  } else {
+    injectCode = `mod.render = function(){}; wx.createComponent(mod)`
+  }
+  const jsRet = [`import mod from ${request}; `, injectCode, `export default mod; export * from ${request}`]
+  return jsRet.join('\n')
 }
